expression/builtin_encryption_vec.go

```diff
@@ -171,11 +175,35 @@ func (b *builtinPasswordSig) vecEvalString(input *chunk.Chunk, result *chunk.Col
 }
 
 func (b *builtinSHA1Sig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSHA1Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+	hasher := sha1.New()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		_, err = hasher.Write(hack.Slice(str))
```

Comment:
We can use `Column.GetBytes` here.

Reply:
ok, I will fix it now, and then update the performance compare for the new implement.